### PR TITLE
feat: Created watch my pixel form

### LINKED ISSUE #18: [Feat]: Frontend watch-my-pixel application form
### My suggestion

Frontend form telling us what pixel and when to watch and inform user about

- [x] Datetime field, shadcn-vue calendar component should do 
- [x] Temp Username field (we do not have user auth at this point)
- [x] Lan Lat Field (we do not have map component)

Design is not our main problem for now but have in consideration that map will take most of page space. 

### Possible alternatives

_No response_

### PR DESCRIPTION
**Pull request by Kubaryt**
Resolves #18

### CODE REVIEW

File: frontend/components/WatchMyPixelForm.vue [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          <div w="2/3" mt-4 flex gap-10>
+            <input
+              id="lat" v-model="formData.lat" type="text" w="1/2" border-b-3 border-b-gray-700 rounded-sm bg-black
+              placeholder="Latitude" focus="outline-none" autocomplete="off"
+            >
+            <input
+              id="lng" v-model="formData.lng" type="text" w="1/2" border-b-3 border-b-gray-700 rounded-sm bg-black
+              placeholder="Longtitude" focus="outline-none" autocomplete="off"
+            >
+          </div>
```
Comment (wikipop) [inline frontend/components/WatchMyPixelForm.vue:22]: Why we are not using <Input/> component from shadcn?

File: frontend/components/WatchMyPixelForm.vue
Comment (wikipop) [inline frontend/components/WatchMyPixelForm.vue:1]: No error handling and data validation.

**Review by wikipop — CHANGES_REQUESTED** [review]

File: frontend/components/WatchMyPixelForm.vue [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      <div mt-8 flex flex-col items-center text-lg>
+        <form flex flex-col items-center w="15svw" @submit.prevent="submitForm">
+          <FormField name="username" :validate-on-blur="!isFieldDirty">
+            <FormItem w="full">
+              <FormControl>
+                <Input v-model="formData.name" type="text" placeholder="Name" !w="full" />
+              </FormControl>
+            </FormItem>
+          </FormField>
+          <FormField name="username" :validate-on-blur="!isFieldDirty">
```
Comment (wikipop) [inline frontend/components/WatchMyPixelForm.vue:16]: double username field

File: frontend/components/WatchMyPixelForm.vue [hunk 3]
```diff
@@ -0,0 +1,91 @@
+<template>
+  <div h-screen w="25svw" flex justify-center bg-black>
+    <div mx-auto mt-32 max-w-full flex flex-col items-center text-left text-gray-300>
+      <div text-center text-4xl>
+        Watch my pixel form
+      </div>
+      <div mt-8 flex flex-col items-center text-lg>
+        <form flex flex-col items-center w="15svw" @submit.prevent="submitForm">
+          <FormField name="username" :validate-on-blur="!isFieldDirty">
```
Comment (wikipop) [inline frontend/components/WatchMyPixelForm.vue:9]: double username field

File: frontend/components/WatchMyPixelForm.vue [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+const formSchema = toTypedSchema(z.object({
+  username: z.string().min(2).max(50)
+}));
+
+const { isFieldDirty, handleSubmit } = useForm({
+  validationSchema: formSchema
+});
+
+const submitForm = handleSubmit((values) => {
+  toast({
```
Comment (wikipop) [inline frontend/components/WatchMyPixelForm.vue:86]: Toast does not seem to work

File: frontend/components/WatchMyPixelForm.vue (hunk 2 above)
Comment (Kubaryt) [reply frontend/components/WatchMyPixelForm.vue]: its calendar field with wrong name, fixed it

**Review by wikipop — APPROVED** [review]

**Review by Norbiros — CHANGES_REQUESTED** [review]
- Resolve conflicts
- Resolve conversations
- I get: Expected number, received string

**Review by Norbiros — CHANGES_REQUESTED** [review]
This code is not perfect, and probably will be changed but it's okay for now.
Before merging fix issues/warns in browser:
```
[Vue warn]: Extraneous non-props attributes (id, aria-describedby, aria-invalid) were passed to component but could not be automatically inherited because component renders fragment or text root nodes. 
  at <PopperRoot id="radix-2-form-item" aria-describedby="radix-2-form-item-description" aria-invalid=false > 
  at <PopoverRoot onUpdate:open=fn id="radix-2-form-item" aria-describedby="radix-2-form-item-description"  ... > 
  at <Popover id="radix-2-form-item" aria-describedby="radix-2-form-item-description" aria-invalid=false > 
  at <PrimitiveSlot id="radix-2-form-item" aria-describedby="radix-2-form-item-description" aria-invalid=false > 
  at <FormControl > 
  at <FormItem w="full" flex="" flex-col=""  ... > 
  at <Field name="date" validate-on-blur=false > 
  at <WatchMyPixelForm > 
  at <Test onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< undefined > > 
  at <Anonymous key="/test" vnode= 
```
And in console:
```
 WARN  [Vue warn]: Extraneous non-props attributes (data-v-inspector, id, aria-describedby, aria-invalid) were passed to component but could not be automatically inherited because component renders fragment or text root nodes. 
  at <PopperRoot data-v-inspector="components/ui/form/FormControl.vue:2:3" id="radix-2-form-item" aria-describedby="radix-2-form-item-description"  ... > 
  at <PopoverRoot onUpdate:open=fn data-v-inspector="components/ui/form/FormControl.vue:2:3" id="radix-2-form-item"  ... > 
  at <Popover id="radix-2-form-item" aria-describedby="radix-2-form-item-description" aria-invalid=false  ... > 
  at <PrimitiveSlot id="radix-2-form-item" aria-describedby="radix-2-form-item-description" aria-invalid=false  ... > 
  at <FormControl > 
  at <FormItem w="full" flex="" flex-col=""  ... > 
  at <Field name="date" validate-on-blur=false data-v-inspector="components/WatchMyPixelForm.vue:17:11" > 
  at <WatchMyPixelForm data-v-inspector="pages/test.vue:7:3" > 
  at <Test onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< undefined > > 
  at <Anonymous key="/test" vnode= { __v_isVNode: true,
  __v_skip: true,
  type:
   { __name: 'test',
     setup: [Function (anonymous)],
     ssrRender: [Function: _sfc_ssrRender],
     __file: '/home/norbi/Development/SideProjects/LandsatWebTracker/frontend/pages/test.vue' },
  props:
   { onVnodeUnmounted: [Function: onVnodeUnmounted],
     ref:
      RefImpl {
        dep: [Dep],
        __v_isRef: true,
        __v_isShallow: false,
        _rawValue: undefined,
        _value: undefined } },
  key: null,
  ref:
   { i:
      { uid: 5,
        vnode: [Object],
        type: [Object],
        parent: [Object],
        appContext: [Object],
        root: [Object],
        next: null,
        subTree: [Object],
        effect: null,
        update: null,
        job: null,
        scope: [EffectScope],
        render: [Function (anonymous)],
        proxy: [Object],
        exposed: null,
        exposeProxy: null,
        withProxy: null,
        provides: [Object <Complex prototype>],
        ids: [Array],
        accessCache: [Object: null prototype] {},
        renderCache: [],
        components: null,
        directives: null,
        propsOptions: [Array],
        emitsOptions: null,
        emit: [Function: bound emit],
        emitted: null,
        propsDefaults: [Object: null prototype] {},
        inheritAttrs: false,
        ctx: [Object],
        data: {},
        props: [Object],
        attrs: {},
        slots: [Object],
        refs: {},
        setupState: {},
        setupContext: [Object],
        suspense: null,
        suspenseId: 0,
        asyncDep: null,
        asyncResolved: false,
        isMounted: false,
        isUnmounted: false,
        isDeactivated: false,
        bc: null,
        c: null,
        bm: null,
        m: null,
        bu: null,
        u: null,
        um: null,
        bum: null,
        da: null,
        a: null,
        rtg: null,
        rtc: null,
        ec: null,
        sp: null },
     r:
      RefImpl {
        dep: [Dep],
        __v_isRef: true,
        __v_isShallow: false,
        _rawValue: undefined,
        _value: undefined },
     k: undefined,
     f: false },
  scopeId: null,
  slotScopeIds: null,
  children: null,
  component: null,
  suspense: null,
  ssContent: null,
  ssFallback: null,
  dirs: null,
  transition: null,
  el: null,
  anchor: null,
  target: null,
  targetStart: null,
  targetAnchor: null,
  staticCount: 0,
  shapeFlag: 4,
  patchFlag: 0,
  dynamicProps: null,
  dynamicChildren: null,
  appContext: null,
  ctx:
   <ref *1> {
     uid: 5,
     vnode:
      { __v_isVNode: true,
        __v_skip: true,
        type: [Object],
        props: [Object],
        key: null,
        ref: null,
        scopeId: null,
        slotScopeIds: null,
        children: [Object],
        component: [Circular *1],
        suspense: null,
        ssContent: null,
        ssFallback: null,
        dirs: null,
        transition: null,
        el: null,
        anchor: null,
        target: null,
        targetStart: null,
        targetAnchor: null,
        staticCount: 0,
        shapeFlag: 36,
        patchFlag: 0,
        dynamicProps: null,
        dynamicChildren: null,
        appContext: null,
        ctx: [Object] },
     type:
      { name: 'RouterView',
        inheritAttrs: false,
        props: [Object],
        compatConfig: [Object],
        setup: [Function: setup] },
     parent:
      { uid: 4,
        vnode: [Object],
        type: [Object],
        parent: [Object],
        appContext: [Object],
        root: [Object],
        next: null,
        subTree: [Object],
        effect: null,
        update: null,
        job: null,
        scope: [EffectScope],
        render: [Function (anonymous)],
        proxy: [Object],
        exposed: [Object],
        exposeProxy: null,
        withProxy: null,
        provides: [Object <Complex prototype>],
        ids: [Array],
        accessCache: [Object: null prototype] {},
        renderCache: [],
        components: null,
        directives: null,
        propsOptions: [Array],
        emitsOptions: null,
        emit: [Function: bound emit],
        emitted: null,
        propsDefaults: [Object: null prototype] {},
        inheritAttrs: false,
        ctx: [Object],
        data: {},
        props: [Object],
        attrs: {},
        slots: {},
        refs: {},
        setupState: {},
        setupContext: [Object],
        suspense: null,
        suspenseId: 0,
        asyncDep: null,
        asyncResolved: false,
        isMounted: false,
        isUnmounted: false,
        isDeactivated: false,
        bc: null,
        c: null,
        bm: null,
        m: null,
        bu: null,
        u: null,
        um: null,
        bum: null,
        da: null,
        a: null,
        rtg: null,
        rtc: null,
        ec: null,
        sp: null },
     appContext:
      { app: [Object],
        config: [Object],
        mixins: [],
        components: [Object],
        directives: {},
        provides: [Object: null prototype],
        optionsCache: [WeakMap],
        propsCache: [WeakMap],
        emitsCache: [WeakMap] },
     root:
      <ref *2> {
        uid: 0,
        vnode: [Object],
        type: [Object],
        parent: null,
        appContext: [Object],
        root: [Circular *2],
        next: null,
        subTree: null,
        effect: null,
        update: null,
        job: null,
        scope: [EffectScope],
        render: [Function: NOOP],
        proxy: {},
        exposed: {},
        exposeProxy: null,
        withProxy: null,
        provides: [Object <Complex prototype>],
        ids: [Array],
        accessCache: [Object: null prototype] {},
        renderCache: [],
        components: null,
        directives: null,
        propsOptions: [],
        emitsOptions: null,
        emit: [Function: bound emit],
        emitted: null,
        propsDefaults: [Object: null prototype] {},
        inheritAttrs: undefined,
        ctx: {},
        data: {},
        props: {},
        attrs: {},
        slots: {},
        refs: {},
        setupState: [Object],
        setupContext: [Object],
        suspense: null,
        suspenseId: 0,
        asyncDep: null,
        asyncResolved: false,
        isMounted: false,
        isUnmounted: false,
        isDeactivated: false,
        bc: null,
        c: null,
        bm: null,
        m: null,
        bu: null,
        u: null,
        um: null,
        bum: null,
        da: null,
        a: null,
        rtg: null,
        rtc: null,
        ec: [Array],
        sp: null,
        devtoolsRawSetupState: [Object] },
     next: null,
     subTree:
      { __v_isVNode: true,
        __v_skip: true,
        type: [Object],
        props: [Object],
        key: null,
        ref: null,
        scopeId: null,
        slotScopeIds: null,
        children: [Object],
        component: null,
        suspense: null,
        ssContent: [Object],
        ssFallback: [Object],
        dirs: null,
        transition: null,
        el: null,
        anchor: null,
        target: null,
        targetStart: null,
        targetAnchor: null,
        staticCount: 0,
        shapeFlag: 160,
        patchFlag: 0,
        dynamicProps: null,
        dynamicChildren: null,
        appContext: null,
        ctx: [Object] },
     effect: null,
     update: null,
     job: null,
     scope:
      EffectScope {
        detached: true,
        _active: true,
        effects: [],
        cleanups: [],
        _isPaused: false,
        parent: undefined },
     render: [Function (anonymous)],
     proxy: { name: [Getter/Setter], route: [Getter/Setter] },
     exposed: null,
     exposeProxy: null,
     withProxy: null,
     provides:
      Object <Object <Complex prototype>> {
        [Symbol(router view depth)]: [ComputedRefImpl],
        [Symbol(router view location matched)]: [ComputedRefImpl],
        [Symbol(router view location)]: [ComputedRefImpl] },
     ids: [ '', 0, 0 ],
     accessCache: [Object: null prototype] {},
     renderCache: [],
     components: null,
     directives: null,
     propsOptions: [ [Object], [Array] ],
     emitsOptions: null,
     emit: [Function: bound emit],
     emitted: null,
     propsDefaults: [Object: null prototype] {},
     inheritAttrs: false,
     ctx: { name: [Getter/Setter], route: [Getter/Setter] },
     data: {},
     props: { route: undefined, name: 'default' },
     attrs: {},
     slots: { default: [Function] },
     refs: {},
     setupState: {},
     setupContext: { attrs: [Getter], slots: [Getter], emit: [Getter], expose: [Function: expose] },
     suspense: null,
     suspenseId: 0,
     asyncDep: null,
     asyncResolved: false,
     isMounted: false,
     isUnmounted: false,
     isDeactivated: false,
     bc: null,
     c: null,
     bm: null,
     m: null,
     bu: null,
     u: null,
     um: null,
     bum: null,
     da: null,
     a: null,
     rtg: null,
     rtc: null,
     ec: null,
     sp: null } } route= { fullPath: '/test',
  hash: '',
  query: {},
  name: 'test',
  path: '/test',
  params: {},
  matched:
   [ { path: '/test',
       redirect: undefined,
       name: 'test',
       meta: {},
       aliasOf: undefined,
       beforeEnter: undefined,
       props: [Object],
       children: [],
       instances: {},
       leaveGuards: Set(0) {},
       updateGuards: Set(0) {},
       enterCallbacks: {},
       components: [Object] } ],
  meta: {},
  redirectedFrom: undefined,
  href: '/test' }  ... > 
  at <RouterView name=undefined route=undefined > 
  at <NuxtPage > 
  at <NuxtLayoutProvider layoutProps= { 'data-v-inspector': 'app.vue:3:5',
  ref:
   RefImpl {
     dep:
      Dep {
        computed: undefined,
        version: 0,
        activeLink: undefined,
        subs: undefined,
        target: undefined,
        map: undefined,
        key: undefined,
        subsHead: undefined },
     __v_isRef: true,
     __v_isShallow: false,
     _rawValue: undefined,
     _value: undefined } } key="default" name="default"  ... > 
  at <NuxtLayout data-v-inspector="app.vue:3:5" > 
  at <App data-v-inspector="../node_modules/.pnpm/nuxt@3.13.1_@parcel+watcher@2.4.1_@types+node@22.5.4_eslint@8.57.0_ioredis@5.4.1_magicast@0.3_a5jktaawzporhlpsakqttb27li/node_modules/nuxt/dist/app/components/nuxt-root.vue:16:5" > 
  at <NuxtRoot>


 WARN  [nuxt] Failed to stringify dev server logs. Received DevalueError: Cannot stringify a function. You can define your own reducer/reviver for rich types following the instructions in https://nuxt.com/docs/api/composables/use-nuxt-app#payload.


```

**Comment by Norbiros** [comment]
Warnings are caused by https://github.com/nuxt/nuxt/issues/29002, so we have to wait for fix in `radix-vue`

**Review by Norbiros — DISMISSED** [review]
It's not perfect but ok for now...

**Review by Norbiros — APPROVED** [review]

**Pull request merged**